### PR TITLE
Add changelog for ICU4X 2.3

### PR DESCRIPTION
**Pull request by sffc**
Generated with the tool and then hand-edited (see commit history).

## Changelog

N/A

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,215 @@
...
+    - Refactor single display name constructors into Tiny, Light, and Heavy data tiers (unicode-org#8233)
+      - Constructors on `RegionDisplayNameOwned`, `ScriptDisplayNameOwned`, `VariantDisplayNameOwned`, and `LanguageIdentifierDisplayNameOwned` are overhauled
+  - `icu_experimental/unicodeset`
+    - Add support for `:EastAsianWidth=:` to unicode set parser (unicode-org#7896)
+  - `icu_experimental/units`
+    - `Convertibles`s are now passed by value, and for `Ratio<BigInt>` the `Convertible` impl is now on the reference (unicode-org#8073)
+    - The `Convertibles` trait was completely overhauled to allow for more accurate calculations (unicode-org#8073)
+    - Remove `UnitsFormatter`, use `CategorizedUnitsFormatter` (unicode-org#8236)
+  - `icu_list`
+    - Prevent correctness errors from out-of-range indices in list patterns. (unicode-org#7887)
```
Comment (robertbastian) [inline CHANGELOG.md]: this was a refactoring only, not behaviour change

```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,215 @@
...
+    - Fix extended year calculations in Gregorian-like and Coptic-like calendars (unicode-org#7849)
+    - Add `Julian` to `AnyCalendar` (unicode-org#7224)
+    - Fix safety issue in `Japanese::try_new_with_buffer_provider` (unicode-org#8095)
+    - Add `AnyCalendarKind::try_new` and deprecate  `AnyCalendarKind::new`. The new version uses locale data to infer calendars from locales. (unicode-org#8102)
+    - Deprecate `CalendarPreferences::resolve_calendar`. This method did not perform likely-subtags expansion. (unicode-org#8102)
+  - `icu_casemapping`
+    - Fix `TrailingCase::Unchanged` handling for Dutch (unicode-org#7863)
+  - `icu_collator`
+    - Add `CollatorBorrowed::new_root` (unicode-org#7893)
+    - Changes to the `CollationSpecialPrimariesV1` data struct (unicode-org#7872)
```
Comment (robertbastian) [inline CHANGELOG.md]: is this changelog-worthy? I don't think the changes were breaking?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,215 @@
...
+    - `impl Hash for CodePointInversionList` (unicode-org#8282)
+  - `icu_datetime`
+    - Support numbering system overrides for datetime patterns when found in data (unicode-org#7905)
+    - Correctly handle CLDR 48.2 data with "c" in the skeletons. (unicode-org#7945)
+    - Correctly handle the `B` pattern in hour field sets (unicode-org#7971)
+    - Correctly implement numeric month fields for calendars with leap months (unicode-org#7986)
+    - Implement data structures for packed range patterns. (unicode-org#7966)
+    - Use the correct calendar even if the region is only implied by the language (i.e. `fa`) (unicode-org#8102)
+    - Add scaffolding for range formatting (unicode-org#8173)
+      - New associated type: `TypedDateDataMarkers::RangeSkel`
```
Comment (robertbastian) [inline CHANGELOG.md]: probably too much detail

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,215 @@
...
+    - Preserve zero minutes in MinuteOptional format for 24-hour hour cycles (unicode-org#8237)
+      - Modified variants on `TimeGranularity`: `Hours23OrNone`, `Hours12`
```
Comment (robertbastian) [inline CHANGELOG.md]: I don't think these were actually renamed?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,215 @@
...
+        - Removed `Writeable` and `TryWriteable` from `LanguageIdentifierDisplayNameOwned` (use `.as_borrowed()` to format).
+      - Constructor behavior: The constructors `try_new()`, `try_new_with_buffer_provider()`, and `try_new_unstable()` no longer fail when the display name is not found.
+    - New constructors `LanguageIdentifierDisplayNameOwned::try_new_menu[_unstable|_with_buffer_provider]` (unicode-org#8216)
+    - New methods: `LanguageIdentifierDisplayNameOwned::try_new_short[_unstable|_with_buffer_provider]`, `LanguageIdentifierDisplayNameOwned::try_new_long[_unstable|_with_buffer_provider]` `LanguageIdentifierDisplayNameOwned::try_new_short_menu[_unstable|_with_buffer_provider]` (unicode-org#8219)
+    - Refactor single display name constructors into Tiny, Light, and Heavy data tiers (unicode-org#8233)
+      - Constructors on `RegionDisplayNameOwned`, `ScriptDisplayNameOwned`, `VariantDisplayNameOwned`, and `LanguageIdentifierDisplayNameOwned` are overhauled
+  - `icu_experimental/unicodeset`
+    - Add support for `:EastAsianWidth=:` to unicode set parser (unicode-org#7896)
```
Comment (robertbastian) [inline CHANGELOG.md]: this no longer lives in experimental

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,215 @@
...
+    - Make `DataLocale::try_from_[str|utf8]` const (unicode-org#8305)
+    - Add `data_locale!` macro (unicode-org#8305)
+  - `icu_locale_fallback`
+    - New crate splitting fallback functionality out of `icu_locale`, including: (unicode-org#8245)
+      - Struct `LocaleFallbacker`
+      - Struct `LocaleFallbackerBorrowed`
+      - Struct `LocaleFallbackerWithConfig`
+      - Struct `LocaleFallbackIterator`
+    - New fn `LocaleFallbackerWithConfig::config` (unicode-org#8305)
+  - `icu_pattern`
```
Comment (robertbastian) [inline CHANGELOG.md]: if you list it under components you should point out that it's a breaking change

```suggestion
  - `icu_pattern` (v0.4 -> v0.5)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,215 @@
...
+    - Expose `CanonicalCombiningClass`' integer field (unicode-org#7997)
+    - Deprecate `HangulSyllableType::LeadingVowelSyllable`, `HangulSyllableType::LeadingVowelTrailingSyllable`, `Script::Ethiopian`, and `Script::Nastaliq` in favor of names matching their official Unicode names (unicode-org#8016)
+      - New associated constants: `HangulSyllableType::LVSyllable`, `HangulSyllableType::LVTSyllable`, `Script::Ethiopic`, `Script::ArabicNastaliq`
+    - Added `Script` constants for some non-Unicode scripts (unicode-org#8017)
+    - Add enumerated property constants for short names and aliases (i.e. `LineBreak::HH`) (unicode-org#8040)
+  - `icu_segmenter`
+    - Add unstable Unihan radical provider data and baked support (unicode-org#7805)
+      - New types: `icu_segmenter::provider::UnihanIrgData<'data>`, `icu_segmenter::provider::SegmenterUnihanRadicalV1`
+      - New associated const: `icu_segmenter::provider::Baked::SINGLETON_SEGMENTER_UNIHAN_RADICAL_V1`
+      - The `experimental_segmenter` example now uses `radaboost` for the Chinese radical model and adds `thadaboost` for Thai
```
Comment (robertbastian) [inline CHANGELOG.md]: this is just data for now, which is left unstable on purpose and not changelog worthy

```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,215 @@
...
+    - Use grapheme segmentation for `LineBreakStrictness:Anywhere` (unicode-org#7941)
+    - add experimental `neo::LineSegmenter` (unicode-org#7942)
```
Comment (robertbastian) [inline CHANGELOG.md]:
```suggestion
    - add experimental `_neo_` constructors (unicode-org#7962)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,215 @@
...
+    - Align word segmenter behaviour with ICU4C and UAX#29 (unicode-org#7952)
+    - Fix a bug in loose line breaking (unicode-org#8111)
+    - Fix rewinding behavior in dictionary segmenter (unicode-org#8195)
+    - Add `WordSegmenter[Borrowed]::load_auto[_unstable|_with_buffer_provider]` to load the complex segmentation data that is loaded by `WordSegmenter::new_auto` (unicode-org#8299)
+- Data model and providers
+  - `icu4x-datagen`
+    - Add `--alt-variant` CLI flag to enable alt variants during datagen. (unicode-org#8025)
+  - `icu_provider`
+    - allow slashes in DataMarkerAttributes (unicode-org#7890)
+    - Add type DataPayloadOr (note: it is already in the 1.5.x changelog) (unicode-org#8163)
```
Comment (robertbastian) [inline CHANGELOG.md]: then remove? definitely remove the note

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,215 @@
...
+  - `icu_provider_fs`
+    - Add defense-in-depth against path traversal. (unicode-org#7887)
+  - `icu_provider_registry`
+    - Export `icu_segmenter::provider::SegmenterUnihanRadicalV1` (unicode-org#7805)
```
Comment (robertbastian) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,215 @@
...
+  - `icu_provider_source`
+    - Add Unihan radical trie generation for `icu_segmenter::provider::SegmenterUnihanRadicalV1` (unicode-org#7805)
+      - `SourceDataProvider` can now load this marker from Unihan IRG data
```
Comment (robertbastian) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,215 @@
...
+    - Deprecated the unihan data source (unicode-org#7882)
+    - Compute properties directly from the `unicode` data source, instead of from `icuexport` (unicode-org#7904)
+    - Warn on unknown alt variants in display names (unicode-org#8010)
+      - Added warnings for unknown `alt` variants in `language.rs`, `script.rs`, and `variant.rs`.
+      - Ignored `-alt-variant` and `-alt-chagos` in `region.rs`.
+    - Add `with_alt_variants` to `SourceDataProvider` to support alt variants. (unicode-org#8025)
+      - New enum: `AltVariantKind`
+      - New method: `SourceDataProvider::with_alt_variants`
+    - Updated `LocaleNamesVariantMediumV1` datagen to normalize variant keys to lowercase and perform case-insensitive lookups. (unicode-org#8084)
```
Comment (robertbastian) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,215 @@
...
+  - `litemap`
+    - Make return types of `LiteMap` iter methods (`iter`, `iter_mut`, `values`, `keys`) concrete: (unicode-org#8072)
+      - New types: `ValuesIter`, `KeysIter`, implements relevant traits, returned by `values` and `keys` respectively.
+      - Changed: The return types of the aforementioned methods go from an `impl DoubleEndedIterator` to concrete types such as `S::KeyValueIter`, `S::KeyValueIterMut` or the aforementioned new types.
+  - `potential_utf`
+    - Use `Box::from_raw()` instead of `transmute` for converting unsized transparent boxes. (unicode-org#7871)
+  - `resb`
+    - Add defense-in-depth around checked multiplication. (unicode-org#7887)
+    - (breaking) APIs produce a typed `I32Pair` instead of `(i32, i32)` for maximum layout soundness (unicode-org#8008)
+    - (breaking) `cast_bytes_to_slice` replaced with type-specific cast functions (unicode-org#8008)
```
Comment (robertbastian) [inline CHANGELOG.md]: do we actually specifically want to mark breaking changes?

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,215 @@
...
+    - New `impl_writeable_delegate!` macro to delegate `Writeable` implementations (unicode-org#8139)
+    - New `impl_try_writeable_delegate!` macro to delegate `TryWriteable` implementations (unicode-org#8139)
+    - Support for `where` clause in `impl_display_with_writeable!` (unicode-org#8139)
+    - Add standard derives and `repr(transparent)` to `LossyWrap` (unicode-org#8132)
+      - `LossyWrap<T>` now derives `Clone`, `Copy`, `PartialEq`, `Eq`, `PartialOrd`, `Ord`, and `Hash`, and is marked `#[repr(transparent)]`, enabling its use inside `Copy` structs and with delegate macros.
+      - Added error mapping fn to `impl_try_writeable_delegate!`
+    - add `TryWriteable::try_writeable_borrow` (unicode-org#8192)
+    - new struct `writeable::adapters::Replace` (unicode-org#8238)
+  - `zerofrom`
+    - enable all features in playground (unicode-org#7958)
```
Comment (robertbastian) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md (hunk 13 above)
Comment (sffc) [reply CHANGELOG.md]: it doesn't _hurt_

File: CHANGELOG.md (hunk 14 above)
Comment (sffc) [reply CHANGELOG.md]: why not? The crate did get released with this change

File: CHANGELOG.md (hunk 1 above)
Comment (sffc) [reply CHANGELOG.md]: 5839b46a01c2515588a963f2bf87370f726f94f9

File: CHANGELOG.md (hunk 7 above)
Comment (sffc) [reply CHANGELOG.md]: It's worth including an entry for this (I did a whole talk about it at UTW) but I changed the wording to set expectations.

File: CHANGELOG.md (hunk 8 above)
Comment (sffc) [reply CHANGELOG.md]: Changed the text but not the PR link since these all link to PRs. I edited the PR description to link to the issue.

File: CHANGELOG.md (hunk 9 above)
Comment (sffc) [reply CHANGELOG.md]: 5839b46a01c2515588a963f2bf87370f726f94f9

File: CHANGELOG.md (hunk 10 above)
Comment (sffc) [reply CHANGELOG.md]: 5839b46a01c2515588a963f2bf87370f726f94f9

File: CHANGELOG.md (hunk 11 above)
Comment (sffc) [reply CHANGELOG.md]: 5839b46a01c2515588a963f2bf87370f726f94f9

File: CHANGELOG.md (hunk 12 above)
Comment (sffc) [reply CHANGELOG.md]: 5839b46a01c2515588a963f2bf87370f726f94f9

File: CHANGELOG.md (hunk 2 above)
Comment (sffc) [reply CHANGELOG.md]: eh, on the fence on this one. It seems worth including, but toward the bottom of the list

File: CHANGELOG.md (hunk 3 above)
Comment (sffc) [reply CHANGELOG.md]: the scaffolding traits are implementable unstably. Unstable clients need the changelog, too, even more than stable clients

File: CHANGELOG.md (hunk 4 above)
Comment (sffc) [reply CHANGELOG.md]: e8e71c5fe42eac2db7959d9ff05479eedf4dff1b

File: CHANGELOG.md (hunk 13 above)
Comment (robertbastian) [reply CHANGELOG.md]: ok, but then: do we do this consistently?

File: CHANGELOG.md (hunk 8 above)
Comment (robertbastian) [reply CHANGELOG.md]: there's no single PR to point at here. unless you want to list ~20 PRs that together describe this line, just link to the issue

File: CHANGELOG.md (hunk 13 above)
Comment (robertbastian) [reply CHANGELOG.md]: for example: we don't do it for the unstable datetime traits that have breaking changes. this seems way more important than marking breaking changes where we bump the major version anyway

File: CHANGELOG.md (hunk 14 above)
Comment (robertbastian) [reply CHANGELOG.md]: it's not a change that a user of `zerofrom` cares about. the use case for the changelog is someone updating their dependencies to the next version, wanting to know what code they can/must change. this is not that. there's a lot of stuff in here that's not that

File: CHANGELOG.md [hunk 15]
```diff
@@ -87,25 +86,28 @@
     - New methods: `LanguageIdentifierDisplayNameOwned::try_new_short[_unstable|_with_buffer_provider]`, `LanguageIdentifierDisplayNameOwned::try_new_long[_unstable|_with_buffer_provider]` `LanguageIdentifierDisplayNameOwned::try_new_short_menu[_unstable|_with_buffer_provider]` (unicode-org#8219)
     - Refactor single display name constructors into Tiny, Light, and Heavy data tiers (unicode-org#8233)
       - Constructors on `RegionDisplayNameOwned`, `ScriptDisplayNameOwned`, `VariantDisplayNameOwned`, and `LanguageIdentifierDisplayNameOwned` are overhauled
+    - Rename single display name types to follow new convention (unicode-org#8319)
+      - New types: `LanguageIdentifierDisplayNameBorrowed`, `RegionDisplayNameBorrowed`, `ScriptDisplayNameBorrowed`, `VariantDisplayNameBorrowed` (borrowed versions of display names)
+      - Renamed types: `LanguageIdentifierDisplayNameOwned` -> `LanguageIdentifierDisplayName`, `RegionDisplayNameOwned` -> `RegionDisplayName`, `ScriptDisplayNameOwned` -> `ScriptDisplayName`, `VariantDisplayNameOwned` -> `VariantDisplayName` (owned versions of display names, formerly had `Owned` suffix)
```
Comment (robertbastian) [inline CHANGELOG.md]: these types were never released, so it's not a rename

File: CHANGELOG.md [hunk 16]
```diff
@@ -132,29 +134,27 @@
     - Added `Script` constants for some non-Unicode scripts (unicode-org#8017)
     - Add enumerated property constants for short names and aliases (i.e. `LineBreak::HH`) (unicode-org#8040)
   - `icu_segmenter`
-    - Add unstable Unihan radical provider data and baked support (unicode-org#7805)
-      - New types: `icu_segmenter::provider::UnihanIrgData<'data>`, `icu_segmenter::provider::SegmenterUnihanRadicalV1`
-      - New associated const: `icu_segmenter::provider::Baked::SINGLETON_SEGMENTER_UNIHAN_RADICAL_V1`
+    - Add `radaboost` and `thadaboost` models for faster and more accurate Chinese and Thai segmentation as unstable, for testing purposes only (unicode-org#7805)
```
Comment (robertbastian) [inline CHANGELOG.md]: but it's not adding the models for testing purposes. clients can't use them for testing purposes. it adds the data and writes a test around the data. this is not useful for clients

File: CHANGELOG.md [hunk 17]
```diff
@@ -132,29 +134,27 @@
...
     - Fix rewinding behavior in dictionary segmenter (unicode-org#8195)
     - Add `WordSegmenter[Borrowed]::load_auto[_unstable|_with_buffer_provider]` to load the complex segmentation data that is loaded by `WordSegmenter::new_auto` (unicode-org#8299)
+    - Add unstable `LineSegmenter::new_17_for_non_complex_scripts`, implementing Unicode 17 (unicode-org#8041)
 - Data model and providers
   - `icu4x-datagen`
     - Add `--alt-variant` CLI flag to enable alt variants during datagen. (unicode-org#8025)
   - `icu_provider`
     - allow slashes in DataMarkerAttributes (unicode-org#7890)
-    - Add type DataPayloadOr (note: it is already in the 1.5.x changelog) (unicode-org#8163)
+    - Stabilize type DataPayloadOr, added as draft in 1.5 (unicode-org#8163)
```
Comment (robertbastian) [inline CHANGELOG.md]: it wasn't draft, it was hidden

Comment (robertbastian) [inline CHANGELOG.md]: it wasn't draft, it was hidden. this is a completely new API from the client perspective. so either list all of its methods or none

File: CHANGELOG.md [hunk 18]
```diff
@@ -172,6 +171,7 @@
     - Update supported Dart toolchain for `record_use` to `3.13.0-215.0.dev` (unicode-org#8119)
     - Dart: in the build hook, skip building if code assets are disabled. (unicode-org#8183)
     - Add `libm` as a library input in the Dart linking script for Android. (unicode-org#8199)
+    - add bindings for icu_locale::names (unicode-org#8341)
```
Comment (robertbastian) [inline CHANGELOG.md]: I don't think we ever listed new APIs in FFI, as they're already covered under components

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,5 +1,215 @@
...
+  - `icu_experimental/units`
+    - `Convertibles`s are now passed by value, and for `Ratio<BigInt>` the `Convertible` impl is now on the reference (unicode-org#8073)
+    - The `Convertibles` trait was completely overhauled to allow for more accurate calculations (unicode-org#8073)
+    - Remove `UnitsFormatter`, use `CategorizedUnitsFormatter` (unicode-org#8236)
+  - `icu_locale`
+    - fixes fallback with language-likely script but region-unlikely script, which fixes data loading and generation behavior for locales including  `sr-Cyrl-ME` and `zh-Hans-TW` (unicode-org#7857)
+    - Allow digits as extension singletons as allowed by BCP47, e.g. `-1-foobar` (unicode-org#8019)
+    - Fixes an issue in `LocaleDirectionality` for unknown languages (unicode-org#7918)
+    - `LocaleExpander::maximize` no longer maximizes `und` to `en-Latn-US` (unicode-org#8110)
+    - new module icu_locale::names, originally part of icu_experimental::displaynames (unicode-org#8338)
```
Comment (robertbastian) [inline CHANGELOG.md]:
```suggestion
    - New module `icu_locale::names`, originally part of `icu_experimental::displaynames` (unicode-org#8338)
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,5 +1,215 @@
...
+    - New methods: `LanguageIdentifierDisplayNameOwned::try_new_short[_unstable|_with_buffer_provider]`, `LanguageIdentifierDisplayNameOwned::try_new_long[_unstable|_with_buffer_provider]` `LanguageIdentifierDisplayNameOwned::try_new_short_menu[_unstable|_with_buffer_provider]` (unicode-org#8219)
+    - Refactor single display name constructors into Tiny, Light, and Heavy data tiers (unicode-org#8233)
+      - Constructors on `RegionDisplayNameOwned`, `ScriptDisplayNameOwned`, `VariantDisplayNameOwned`, and `LanguageIdentifierDisplayNameOwned` are overhauled
+    - Rename single display name types to follow new convention (unicode-org#8319)
+      - New types: `LanguageIdentifierDisplayNameBorrowed`, `RegionDisplayNameBorrowed`, `ScriptDisplayNameBorrowed`, `VariantDisplayNameBorrowed` (borrowed versions of display names)
+      - Renamed types: `LanguageIdentifierDisplayNameOwned` -> `LanguageIdentifierDisplayName`, `RegionDisplayNameOwned` -> `RegionDisplayName`, `ScriptDisplayNameOwned` -> `ScriptDisplayName`, `VariantDisplayNameOwned` -> `VariantDisplayName` (owned versions of display names, formerly had `Owned` suffix)
```
Comment (robertbastian) [inline CHANGELOG.md]: all of this should be listed under `icu_locale`

**Comment by robertbastian** [comment]
please split the tool changes from the actual changelog for 2.3

File: CHANGELOG.md (hunk 13 above)
Comment (sffc) [reply CHANGELOG.md]: Consistency is important for stable crates, but partial documentation for unstable crates is still helpful. I don't want to delete these breadcrumbs in the name of consistency.

File: CHANGELOG.md (hunk 14 above)
Comment (sffc) [reply CHANGELOG.md]: They will see that zerofrom had a version bump, though. I think any crate with a version bump should at least have a brief line saying what it was. Saying that it was changing cargo toml metadata is a good signal that the user _doesn't_ need to worry about that version bump.

File: CHANGELOG.md (hunk 16 above)
Comment (sffc) [reply CHANGELOG.md]: The code and data are there for users who want to get at them manually, even if not available through the library interface. There are clients who care a lot about segmentation and are willing to take the extra steps to use the code for testing.

File: CHANGELOG.md (hunk 14 above)
Comment (robertbastian) [reply CHANGELOG.md]: no, they will just assume we fixed a typo in a comment or something. there's nothing to see here

File: CHANGELOG.md (hunk 16 above)
Comment (robertbastian) [reply CHANGELOG.md]: and these kinds of clients should follow the relevant issues in the repo, not find this through the changelog

File: CHANGELOG.md (hunk 20 above)
Comment (sffc) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 19 above)
Comment (sffc) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 18 above)
Comment (sffc) [reply CHANGELOG.md]: I think listing them is good practice.

To preempt your likely next comment: I don't know if these entries are comprehensive, but the ones we have we shouldn't be deleting in the name of consistency.

File: CHANGELOG.md (hunk 17 above)
Comment (sffc) [reply CHANGELOG.md]: Rephrased again, and I put a link to the second PR with the new methods without naming them

File: CHANGELOG.md (hunk 16 above)
Comment (sffc) [reply CHANGELOG.md]: I rearranged the items. it isn't first any more. But I will keep the line item

File: CHANGELOG.md (hunk 15 above)
Comment (sffc) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md (hunk 14 above)
Comment (sffc) [reply CHANGELOG.md]: I don't know why this is controversial. The crate changed versions, so it should be listed in the changelog.

File: CHANGELOG.md (hunk 5 above)
Comment (sffc) [reply CHANGELOG.md]: good catch. I wouldn't have caught this one myself. I moved it to icu_properties.

File: CHANGELOG.md (hunk 6 above)
Comment (sffc) [reply CHANGELOG.md]: I added these.

File: CHANGELOG.md (hunk 8 above)
Comment (sffc) [reply CHANGELOG.md]: ok if you insist

**Comment by sffc** [comment]
> please split the tool changes from the actual changelog for 2.3

The tool changes were already merged. But the github diff didn't reflect that. I landed a merge commit from main which seems to have fixed this.

File: CHANGELOG.md (hunk 3 above)
Comment (sffc) [reply CHANGELOG.md]: I moved this line item under date range formatting

File: CHANGELOG.md (hunk 18 above)
Comment (robertbastian) [reply CHANGELOG.md]: listing half the changes is going to be more confusing for clients that listing none of them

File: CHANGELOG.md (hunk 13 above)
Comment (robertbastian) [reply CHANGELOG.md]: if you mark some breaking changes as `(breaking)`, but others not, readers will rely on the absence of `(breaking)` to mean non-breaking

File: CHANGELOG.md (hunk 14 above)
Comment (robertbastian) [reply CHANGELOG.md]: if you want to list this, put "Misc docs improvements" under General.

the changelog should be actionable. it should not be as concise as possible, and not create extra work and cognitive load for clients. we want to advertise new APIs and relevant API changes. if I read a line like this in a changelog, I would stop reading, it's just a waste of time

File: CHANGELOG.md (hunk 13 above)
Comment (sffc) [reply CHANGELOG.md]: oh your comment is about the literal text "(breaking)"

I kind-of like it. I added it to icu_pattern. I don't think we need it for icu_experimental.

Comment (robertbastian) [reply CHANGELOG.md]: what did you think "specifically mark breaking changes" meant?

> I added it to icu_pattern. I don't think we need it for icu_experimental.

we should do it for changes that are actually breaking. if there's a major semver bump, they are not breaking (icu_pattern, icu_experimental). the only breaking changes we should have are behind unstable features, such as the datetime thing I mentioned earlier

File: CHANGELOG.md (hunk 16 above)
Comment (robertbastian) [reply CHANGELOG.md]: we explicitly landed this code as `cfg(test)` because it's in no shape to be released. we did _not_ land this as `cfg(feature = "unstable")`, which would make it changelog-worthy

Comment (sffc) [reply CHANGELOG.md]: rephrased again

We had these listed in the 2.2 changelog. The only actually new thing in 2.3 is thadaboost.

File: CHANGELOG.md (hunk 13 above)
Comment (sffc) [reply CHANGELOG.md]: ok I removed the (breaking)s

File: CHANGELOG.md (hunk 14 above)
Comment (sffc) [reply CHANGELOG.md]: I want to list it under utils with version bumps, because it is a util with a version bump

I rephrased it to say "cargo metadata" instead of "playground"

Comment (robertbastian) [reply CHANGELOG.md]: not every version bump is changelog-worthy. let's discuss in the meeting

File: CHANGELOG.md (hunk 13 above)
Comment (Manishearth) [reply CHANGELOG.md]: I like `(breaking)`, we've used it before. It's a good way of pointing out what things will make an upgrade no longer easy. We document version diffs so people can easily see that this is not _automatically_ breaking.

File: CHANGELOG.md (hunk 14 above)
Comment (Manishearth) [reply CHANGELOG.md]: I have been consistently handling this for the last 2? 3? changelogs I've done by either saying "General changes only" or "Internal cleanups" in bumped utils. I agree that this is not meaningful to people.

We should include the entry, with a version diff, but we should  be fine just saying it had minor internal changes.

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,5 +1,196 @@
 # Changelog
 
+## icu4x 2.3
+
+- Components
+  - General
+    - Updated data to TZDB 2026c (unicode-org#8200)
+  - `icu_calendar`
+    - Fix extended year calculations in Gregorian-like and Coptic-like calendars (unicode-org#7849)
+    - Add `Julian` to `AnyCalendar` (unicode-org#7224)
```
Comment (Manishearth) [inline CHANGELOG.md:10]: nit: this needs an API diff (since it'll add a Kind variant etc)

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,5 +1,196 @@
...
+    - Deprecate `CalendarPreferences::resolve_calendar`. This method did not perform likely-subtags expansion. (unicode-org#8102)
+  - `icu_casemapping`
+    - Fix `TrailingCase::Unchanged` handling for Dutch (unicode-org#7863)
+  - `icu_collator`
+    - Add `CollatorBorrowed::new_root` (unicode-org#7893)
+    - Fix a possible panic when generating a sort key for a collation that uses backward secondary level. (unicode-org#7927)
+    - Fix generation of identical level sort keys containing the codepoint U+A000. (unicode-org#7928)
+    - Fixed an issue where the emoji collation was not loading correctly (unicode-org#7989)
+    - Tune the performance of sort key generation (unicode-org#7930)
+    - Changes to the `CollationSpecialPrimariesV1` data struct (unicode-org#7872)
```
Comment (Manishearth) [inline CHANGELOG.md]: nit: "Optimize"? "changes" doesn't really tell me much but elicits questions

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,5 +1,196 @@
...
+  - `icu_collections`
+    - Fix a bug in `CodePointInversionList::contains_set` (unicode-org#8121)
+    - `impl Hash for CodePointInversionList` (unicode-org#8282)
```
Comment (Manishearth) [inline CHANGELOG.md]: thought: Maybe we say "New trait impl:" before this?

idk how consistent we want to be here

File: CHANGELOG.md [hunk 24]
```diff
@@ -1,5 +1,196 @@
...
+  - `icu_datetime`
+    - Add unstable range formatter (unicode-org#8149)
+      - New types: `DateRangeFormatter`, `FixedCalendarDateRangeFormatter`, `FormattedDateRange`
```
Comment (Manishearth) [inline CHANGELOG.md]: nit: NoCalendar also

File: CHANGELOG.md [hunk 25]
```diff
@@ -1,5 +1,196 @@
...
+    - Support numbering system overrides for datetime patterns when found in data (unicode-org#7905)
+    - Implement flexible day periods, i.e. the `B` pattern in hour field sets (unicode-org#7971)
+    - Use the correct calendar even if the region is only implied by the language (i.e. `fa`) (unicode-org#8102)
+    - Preserve zero minutes in MinuteOptional format for 24-hour hour cycles (unicode-org#8237)
+    - Correctly implement numeric month fields for calendars with leap months (unicode-org#7986)
+    - Correctly handle CLDR 48.2 data with "c" in the skeletons (unicode-org#7945)
+  - `icu_decimal`
+    - Fix integer overflow in UnsignedDecimal exponent parsing (unicode-org#8289)
+  - `icu_experimental`: `0.5.0 -> 0.6.0`
+  - `icu_experimental/currency`
```
Comment (Manishearth) [inline CHANGELOG.md]: issue: please nest these

File: CHANGELOG.md [hunk 26]
```diff
@@ -1,5 +1,196 @@
...
+    - The `Convertibles` trait was completely overhauled to allow for more accurate calculations (unicode-org#8073)
+    - `Convertibles`s are now passed by value, and for `Ratio<BigInt>` the `Convertible` impl is now on the reference (unicode-org#8073)
+    - Remove `UnitsFormatter`, use `CategorizedUnitsFormatter` (unicode-org#8236)
+  - `icu_locale`
+    - `LocaleExpander::maximize` no longer maximizes `und` to `en-Latn-US` (unicode-org#8110)
+    - Fix an issue in `LocaleDirectionality` for unknown languages (unicode-org#7918)
+    - Fix `LocaleExpander::maximize()` to strip placeholder `Zzzz`/`ZZ` subtags before matching, per UTS #35 (unicode-org#8240)
+    - Fix fallback with language-likely script but region-unlikely script, which fixes data loading and generation behavior for locales including  `sr-Cyrl-ME` and `zh-Hans-TW` (unicode-org#7857)
+    - Allow digits as extension singletons as allowed by BCP47, e.g. `-1-foobar` (unicode-org#8019)
+  - `icu_locale/names`
```
Comment (Manishearth) [inline CHANGELOG.md]: issue: this should just be in `icu_locale`. Fine to nest as `icu_locale::names`

File: CHANGELOG.md [hunk 27]
```diff
@@ -1,5 +1,196 @@
...
+    - Warn on unknown alt variants in display names (unicode-org#8010)
+      - Added warnings for unknown `alt` variants in `language.rs`, `script.rs`, and `variant.rs`.
+      - Ignored `-alt-variant` and `-alt-chagos` in `region.rs`.
+- FFI
+  - General
+    - Fix an issue in JS bindings where enums in objects were not parsed correctly (unicode-org#7885)
+    - Add bindings for `icu_locale::names` (unicode-org#8341)
+unicode-org#8119)
+    - Dart, Kotlin: use the `Script` type on `ScriptExtension` APIs (unicode-org#7996)
+  - Dart
```
Comment (Manishearth) [inline CHANGELOG.md:154]: thought: hopefully https://github.com/unicode-org/icu4x/pull/8371 can make the release

File: CHANGELOG.md [hunk 28]
```diff
@@ -1,5 +1,196 @@
...
+    - Update supported Dart toolchain for `record_use` to `3.13.0-215.0.dev` (    - In the build hook, skip building if code assets are disabled. (unicode-org#8183)
```
Comment (Manishearth) [inline CHANGELOG.md]: issue: missing newline

File: CHANGELOG.md (hunk 27 above)
Comment (Manishearth) [reply CHANGELOG.md:154]: It merged, please add

**Review by Manishearth — APPROVED** [review]
